### PR TITLE
http: implement TLS and InsecureSkipVerify

### PR DESCRIPTION
**Pull request by lausser**
Allow InsecureSkipVerify for self-signed certificates (Issue  #41)

### CODE REVIEW

File: http.go [hunk 1]
```diff
@@ -61,6 +61,16 @@ func probeHTTP(target string, w http.ResponseWriter, module Module) (success boo
 	client := &http.Client{
 		Timeout: module.Timeout,
 	}
+	if module.HTTP.TLS {
```
Comment (brian-brazil) [inline http.go]: There's no need for this, always generate the tls config.

File: http.go [hunk 2]
```diff
@@ -61,6 +61,16 @@ func probeHTTP(target string, w http.ResponseWriter, module Module) (success boo
...
+		config, err := module.HTTP.TLSConfig.GenerateConfig()
+		if err != nil {
+			log.Errorf("Error configtls request for target %s", err)
+			return
```
Comment (brian-brazil) [inline http.go]: Explicitly return false here

File: http.go [hunk 3]
```diff
@@ -61,6 +61,16 @@ func probeHTTP(target string, w http.ResponseWriter, module Module) (success boo
 	client := &http.Client{
 		Timeout: module.Timeout,
 	}
+	if module.HTTP.TLS {
+		config, err := module.HTTP.TLSConfig.GenerateConfig()
+		if err != nil {
+			log.Errorf("Error configtls request for target %s", err)
```
Comment (brian-brazil) [inline http.go]: This reads oddly. "Error generating TLS config"

File: main.go [hunk 4]
```diff
@@ -58,6 +58,8 @@ type HTTPProbe struct {
 	Headers                map[string]string `yaml:"headers"`
 	FailIfMatchesRegexp    []string          `yaml:"fail_if_matches_regexp"`
 	FailIfNotMatchesRegexp []string          `yaml:"fail_if_not_matches_regexp"`
+	TLS                    bool              `yaml:"tls"`
```
Comment (brian-brazil) [inline main.go]: This is no longer needed.

**Comment by lausser** [comment]
I'm not sure if the additional test (http-webserver and insecure_skip_verify=false) is too much, i felt like "even if there's no benefit, at least there is no harm"

**Comment by brian-brazil** [comment]
More tests are always good. Can you squash your commits?

**Comment by syepes** [comment]
+1 A must have ;-)

**Comment by brian-brazil** [comment]
Could you add a insecure_skip_verify: false to the http_2xx example in the readme?

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**